src/services/WhitelistService.ts

```diff
@@ -0,0 +1,39 @@
+import { IFSConfig } from '../interfaces/IFSConfig';
+import { IFTPLogin } from '../interfaces/IFTPLogin';
+import { IWhiteList } from '../interfaces/IWhiteList';
+import { FSService } from '../services/FSService';
+import { FTPService } from '../services/FTPService';
+import { RESTService } from '../services/RESTService';
+import { uploads } from '../../config/config.json';
+
+
+// const fsconfig = JSON.parse(fs.readFileSync('./config.json', 'utf-8'));
+
+export class WhitelistService {
+  service?: FSService | FTPService | RESTService;
+
+  constructor() {
+    this.service = undefined;
+  }
+
+  uploadAll(whitelist: string): Promise<boolean> {
```

Comment:
This is ok, but just FYI this is a school example of where you can eliminate the switch by introducing the Strategy Pattern (Design Pattern) --> https://www.webdevelopment.co.nz/post/kill-the-switch-with-the-strategy-pattern